vault/logical_system.go

```diff
@@ -2744,22 +2742,32 @@ func (b *SystemBackend) handleMonitor(ctx context.Context, req *logical.Request,
 	// a gzip stream it will go ahead and write out the HTTP response header
 	_, err = w.Write([]byte(""))
 	if err != nil {
-		return resp, fmt.Errorf("error seeding flusher: %w", err)
+		return nil, fmt.Errorf("error seeding flusher: %w", err)
 	}
 
 	flusher.Flush()
 
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
 	// Stream logs until the connection is closed.
 	for {
 		select {
+		// Periodically check for the seal status and return if core gets
+		// marked as sealed.
+		case <-ticker.C:
+			if b.Core.Sealed() {
+				// If we error here, it means we couldn't write to the response
+				// writer so there's nothing we can use send over the error.
+				_, _ = fmt.Fprint(w, "core received sealed state change, ending monitor session")
+				return nil, nil
+			}
 		case <-ctx.Done():
-			return resp, nil
+			return nil, nil
 		case l := <-logCh:
-			_, err = fmt.Fprint(w, string(l))
-
-			if err != nil {
-				return resp, err
-			}
+			// If we error here, it means we couldn't write to the response
+			// writer so there's nothing we can use to send over the error.
+			_, _ = fmt.Fprint(w, string(l))
```

Comment:
I think we should probably still return if we see an error though, right?

Reply:
Yeah, I guess we can still return the error. I'll re-word the comment above since that error does indeed get swallowed by this:

https://github.com/hashicorp/vault/blob/c68e270863cd5518a3380b73c721fe822592b6b8/http/handler.go#L823-L833

Reply:
Updated now.